src/workerd/io/io-own.h

```diff
@@ -120,9 +121,20 @@ class DeleteQueue: public kj::AtomicRefcounted {
   DeleteQueue(): crossThreadDeleteQueue(State{kj::Vector<OwnedObject*>()}) {}
 
   void scheduleDeletion(OwnedObject* object) const;
+  void scheduleAction(jsg::Lock& js, kj::Function<void(jsg::Lock&)>&& action) const;
 
   struct State {
     kj::Vector<OwnedObject*> queue;
+    // Collection of actions to be run immediately after pending deletions are processed.
```

Comment:
This comment seems like it's mostly addressed to me, telling me why my idea didn't work. :) I agree, you're right, my idea doesn't work, but probably future readers won't start by thinking of it and so would be better served by a different comment, like:

> Actions that some other IoContext has requested be executed in this IoContext. When adding an action to this list, `crossThreadFulfiller` should be fulfilled, signaling the target IoContext to wake up and run actions. After draining the actions queue, the target IoContext should replace `crossThreadFulfiller` with a new one which will wake it up again.
>
> In particular, these actions are used to implement cross-context promise resolution.
>
> Keep in mind the IoContext could be destroyed before the cross-thread signal runs, in which case the actions will never run.